source/_posts/2020-03-14-2020-using-vyper-with-embark.md

```diff
@@ -0,0 +1,204 @@
+        fromIndex: 0,
+        args: [100]
+      }
+    }
+    // ...
+  }
+}
+```
+Please see the Embark [smart contract configuration documentation](https://framework.embarklabs.io/docs/contracts_configuration.html) for more information on how to configure contracts in Embark.
+
+## Step 5. Run Embark
+Now that we have installed and configured everything, let's run Embark and watch the magic!
+```
+embark run --nodashboard --nobrowser
+```
+Assuming we kept the file name `SimpleStorage.vy` from step 4, we should see the following output in the console:
+```
+compiling Vyper contracts...
+deploying contracts
+Deploying SimpleStorage with 144379 gas at the price of 2000000000 Wei. Estimated cost: 288758000000000 Wei  (txHash: 0x370a864b12b1785b17180b2a10ec2f941a15638eeffadfecf5bbe68755a06f14)
+SimpleStorage deployed at 0x5Baf4D88bC454537C51CEC7568a1E23400483abc using 135456 gas (txHash: 0x370a864b12b1785b17180b2a10ec2f941a15638eeffadfecf5bbe68755a06f14)
+```
+
+### Troubleshooting
+There are a few common issues you may experience when Embark attempts to compile and deploy your Vyper contracts.
+
+#### Vyper is not installed on your machine
+```
+Vyper is not installed on your machine
+You can install it by visiting: https://vyper.readthedocs.io/en/latest/installing-vyper.html
+```
+This means that the binary `vyper` cannot be found on your machine. If you're running *nix, you can verify this by running:
+```
+which vyper
+# expected output: path/to/virtual-env/bin/vyper
+# actual output: vyper not found
+```
+If `vyper` was [installed in a virtual environment](https://vyper.readthedocs.io/en/latest/installing-vyper.html#creating-a-virtual-environment), you may have forgotten to active the environment, ie:
+```
+source path/to/vyper-env/bin/activate
+```
+If `vyper` was not installed in a virual environment, it means either your `PATH` environment variable needs to be updated to point to the directory where the `vyper` binary is installed. You can inspect your `PATH` environment variable by:
+```
+echo $PATH
+```
+And update it with the path to *the directory containing* the `vyper` binary using:
+```
+export PATH=/path/to/vyper/dir:$PATH
+```
+
+#### Error deploying contract
+```
+SimpleStorage has no code associated
+did you mean "simple_storage"?
+deploying contracts
+Error deploying contract simple_storage
+[simple_storage]: Invalid number of parameters for "constructor". Got 0 expected 1!
+Error deploying contracts. Please fix errors to continue.
+```
+This means that the contract `simple_storage` is a file in `contracts/` and expected to be deployed by Embark, but is not configured for deploy in the contracts configuration. This could be the result of changing the filename of the Vyper contract to `simple_storage.vy`, but not updating the contracts configuration, as outlined in [step 4](#Changing-the-contract-filename).
+
+## Step 6. Interact with your contract in Cockpit
+We can use Cockpit to interact with our deployed Vyper contract.
```

Comment:
Optional, but you can also say that the demo will work with the new contract too, because it has the same function names